src/scheduler/uncontrolled_randomness.rs

```diff
@@ -0,0 +1,124 @@
+use crate::runtime::task::TaskId;
+use crate::scheduler::{Schedule, Scheduler};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+enum ScheduleRecord {
+    Task(Option<TaskId>, Vec<TaskId>, bool),
```

Comment:
Probably use `SmallVec<[TaskId; DEFAULT_INLINE_TASKS]>` instead of the Vec, to save an allocation on every step.